### DISCUSSION TITLE
v4.0 邀您共建 🚀

### DISCUSSION THREAD

**Discussion by richard1015** [jdf2e/nutui#1906] [Announcements]
## Task List

>  参与共建 v4 分支  , 贡献者建议每日同步一次主仓库代码

``` bash
#项目启动说明 node 16+ 版本以上 v4分支

pnpm install
pnpm dev

# 修改完成后执行测试命令，验证 ，以及对比线上老版本UI比对  https://nutui.jd.com/demo.html#/cell
pnpm test button
```

#### 工作背景
3.x版本中 在实际开发过程中 会发现 button只是引用了一个很小的loading icon 导致需要全量引用icon ，这会导致开发者的项目文件增大。
我们在v4版本中 为了解决此问题，针对icon组件做了单独抽离，使其可以进行按需加载使用。
在此非常欢迎大家一起共建

> ⚠️ 重要关注  vue中使用 @nutui/icons-vue 小程序 taro 中使用 @nutui/icons-vue-taro

本次任务困难程度分为
- 简单：`Demo文档处理` 参考 PR 4cc2970e267b1863c062a94212114280feb101a6
- 简单：`无需扩展插槽`  参考 PR 4cc2970e267b1863c062a94212114280feb101a6
- 复杂：`需扩展插槽` 参考 PR 8a404f984c96021ba37323e302887ee09e4e28e6
处理完成后提交PR 到 v4 分支

<details open>
<summary>任务列表</summary>

数量 | 组件名称 | 处理方式 | 认领人 | 当前状态 | 计划完成时间
-- | -- | -- | -- | -- | --
1 | Avatar  头像 | 需扩展插槽 | @eiinu  | 已完成 f3c7a7b9c75c1d17285561c8ccf5bd7a62d90f2e  | 2022-12-22
2 | Button  按钮 | 需扩展插槽 | @eiinu  | 已完成 00997c7a7eb4092e38acf6198c834c47ac6bfc99  |  2022-12-20
3 | Cell  单元格 | 无需扩展插槽 | @richard1015 | 已完成 1f125a3caf670d5fd00dabdbb389ac9212ac9f25 |  2022-12-12
4 | Icon  图标 |   | @richard1015 | 已处理 4ec4639349d9b72499d3937f2881440cf3c5411e |  2022-12-12
5 | ImagePreview  图片预览 | 需扩展插槽 |  @yangxiaolu1993  | 已完成 5b21dc258600b6d969ddd63e36185fc3b9935671   |  
6 | ActionSheet  动作面板 | 无需扩展插槽 | @kasonyang  | 已完成 #1928 |  2022-12-14
7 | BackTop  返回顶部 | 无需扩展插槽 | @richard1015   | 已完成 4cc2970e267b1863c062a94212114280feb101a6  |  2022-12-14
8 | Collapse  折叠面板 | 需扩展插槽 |  @Ymm0008  | 已完成 da16f3181626520ceb184296ac268c1b85343cfb  |  2022-12-19
9 | InfiniteLoading  滚动加载 | 需扩展插槽 |  @yangxiaolu1993   | 已完成 5c42bbc57270587c404b70b2b04be17fe91ad731  |  
10 | Popup  弹出层 | 需扩展插槽 | @ailululu  |  已完成 #1953 | 
11 | Steps  步骤条 | 需扩展插槽 |  @eiinu  |  已完成 96727a7a47c47d462aef9e1ff6890f2848211229  | 2022-12-15  
12 | Switch  开关 | 需扩展插槽 | @eiinu  | 已完成 6bc4fa6a4d2711f1c6551fafbfcf3f18bbc71f11 | 2022-12-22
13 | Toast  吐司 | 需扩展插槽 |  @richard1015    | f866bb720738ff6d5e5e080f3ab58560d9887ef8  |  
14 | Progress  进度条 | 需扩展插槽 | @ailululu     |  已完成 #1953 | 
15 | NoticeBar  公告栏 | 需扩展插槽 |  @Ymm0008  | 已完成  |  2022-12-20
16 | SearchBar  搜索栏 | 需扩展插槽 | @Ymm0008   |  已完成  307cd538d81d3f1bff26257c978538564abb79e2 |  2022-12-20
17 | Swiper 轮播图 | Demo 需处理 |  @gyt95  |  已完成 #1937  | 2022-12-16 
18 | Audio音频播放器 | 无需扩展插槽 |  @gyt95 |  已完成 #1930   |  2022-12-15
19 | Navbar  头部导航 | 需扩展插槽 | @ailululu   |  已完成 #1953 | 
20 | FixedNav  悬浮导航 | 无需扩展插槽 | @richard1015 | 已完成 8e95c6874bb3b9a7c1233d1bf393f74b8b83a681  |  2022-12-13
21 | Menu  菜单组件 | 需扩展插槽 | @lkjh3214   |    |  2022-12-22
22 | Tabbar  标签栏 | 需扩展插槽 |  @Drjingfubo  | 已完成 bb714dae9aec0f713f02f1e7fca07c9342c0cc07  |  
23 | Elevator  电梯楼层 | Demo 需处理 | @gyt95   |  已完成 #1936 |  2022-12-16
24 | Tabs  选项卡切换 | 无需扩展插槽 | @thx125  |  已完成 #1927  |  2022-12-14
25 | Checkbox  复选按钮 | 需扩展插槽 |  @lkjh3214  | 已完成 #1964  |  2022-12-21
26 | InputNumber  数字输入框 | 需扩展插槽 | @eiinu  |  已完成 f3d5262a16938c6e6940aef53b839d1870a63fda  | 2022-12-15  
27 | Input  输入框 | 需扩展插槽 | @lkjh3214   | 已完成 #1964  |  2022-12-21
28 | Radio  单选按钮 | 需扩展插槽 | @lkjh3214  | 已完成 #1964  |  2022-12-21
29 | Rate  评分 | 需扩展插槽 | @Ymm0008   | 已完成 #1957  |  
30 | ShortPassword  短密码 | 无需扩展插槽 | @kasonyang  | 已完成 #1938  |  2022-12-14
31 | Uploader  上传 | 需扩展插槽 | @richard1015    |  已完成 8a404f984c96021ba37323e302887ee09e4e28e6  |  2022-12-21
32 | Badge  徽标 | Demo 需处理 |  @Coffee-C  |  已完成 #1948 |  2022-12-18
33 | Tag  标签 | 无需扩展插槽 |  @Coffee-C  |  已完成 #1929   |  2022-12-15
34 | Popover  气泡弹出框 | 需扩展插槽 | @yangxiaolu1993   | 已完成 f14951faf91f31b18b290456b64f82a5a9cd2bc8  |  
35 | Address  地址 | 需扩展插槽 | @yangxiaolu1993   | 已完成 375fcb15cd84bf71ef9086b7ab6cf3bc7a3fbeb0  |  
36 | SideNavBar 侧边栏导航 | 无需扩展插槽 | @gyt95  | 已完成 #1931   |  2022-12-16
37 | Table表格组件 | 无需扩展插槽 | @sjx12  |  已完成 #1947  |  2022-12-18
38 | Grid宫格组件（社区开发者-haiweilian） | 需扩展插槽 | @eiinu  | 已完成 8564ca8fc5a62183c648febce32956d3d98ba001 |  2022-12-14
39 | image | 无需扩展插槽 | @xiaosifeng  |  已完成 #1940  |  2022-12-15
40 | 评论组件 | 无需扩展插槽 | @yangxiaolu1993   | 已完成 2930a64c39094dde67dfc5eccdbaa9383255eafe  |  
41 | TrendArrow 趋势箭头 | 需扩展插槽 | @eiinu  | 已完成 a746b0c41ef0f1b6fceebac249a00a3eb87075c6  |  2022-12-20
42 | pagination 分页 | Demo修改 | @richard1015   | 已完成 3d46443064f2f1199cbdf7c20ca4223ba814d371  |  2022-12-22
43 | cascader 级联 | 无需扩展插槽 | @richard1015   | 已完成 c7d5f45313fd165482aa9fff445ed0a718f879db  |  2022-12-23

</details>

**Comment by kasonyang** [comment]
认领 任务30-ShortPassword 短密码，计划2022-12-15完成

**Reply by richard1015** [reply]
已更新

**Comment by kasonyang** [comment]
认领任务6-ActionSheet 动作面板，计划2022-12-14完成

**Reply by richard1015** [reply]
已更新

**Comment by Kika-Chen** [comment]
认领任务33 - Tag标签，计划 2022-12-15完成

**Reply by richard1015** [reply]
已更新

**Comment by gyt95** [comment]
认领 任务18-Audio 音频播放器，计划 2022-12-15 完成

**Reply by richard1015** [reply]
任务列表已更新

**Comment by thx125** [comment]
认领 任务7 - BackTop 返回顶部，计划2022-12-14完成

**Reply by richard1015** [reply]
不好意思，刚才写了个demo示例，已完成

**Comment by thx125** [comment]
认领 任务24 - Tabs 选项卡切换 计划2022-12-14完成

**Reply by richard1015** [reply]
已更新任务列表

**Comment by sjx12** [comment]
认领 任务24 - Table表格组件 计划2022-12-18完成

**Reply by richard1015** [reply]
已更新

**Comment by gyt95** [comment]
认领 任务36-SideNavBar 侧边栏导航，计划 2022-12-16 完成

**Reply by richard1015** [reply]
已更新

**Comment by xiaosifeng** [comment]
认领 任务39-image，计划2022-12-15完成

**Reply by richard1015** [reply]
已更新

**Comment by gyt95** [comment]
认领 任务23-Elevator 电梯楼层，计划 2022-12-16 完成

**Reply by richard1015** [reply]
已更新

**Comment by gyt95** [comment]
认领 任务17-Swiper 轮播图，计划 2022-12-16 完成

**Reply by eiinu** [reply]
已更新

**Comment by Kika-Chen** [comment]
认领 任务13-Toast 吐司，计划 2022-12-18 完成
认领 任务32-Badge 徽标，计划 2022-12-17 完成

**Reply by richard1015** [reply]
已更新

**Comment by richard1015** [comment]
已发布 v4.0.0 🚀 
https://github.com/jdf2e/nutui/releases/tag/v4.0.0
